aya/src/util.rs

```diff
@@ -13,9 +13,130 @@ use crate::{
     Pod,
 };
 
-use libc::{if_nametoindex, sysconf, _SC_PAGESIZE};
+use libc::{if_nametoindex, sysconf, uname, utsname, _SC_PAGESIZE};
 
-use io::BufRead;
+/// Represents a kernel version, in major.minor.release version.
+// Adapted from https://docs.rs/procfs/latest/procfs/sys/kernel/struct.Version.html.
+#[derive(Debug, Copy, Clone, Eq, PartialEq, PartialOrd)]
+pub struct KernelVersion {
+    pub(crate) major: u8,
+    pub(crate) minor: u8,
+    pub(crate) patch: u16,
+}
+
+impl KernelVersion {
+    /// Constructor.
+    pub fn new(major: u8, minor: u8, patch: u16) -> Self {
+        Self {
+            major,
+            minor,
+            patch,
+        }
+    }
+
+    /// Returns the kernel version of the currently running kernel.
+    pub fn current() -> Result<Self, String> {
```

Comment:
I think we should return a `Box<dyn Error>` instead of String here

Reply:
Can you help me understand how that's better? Currently aya does not depend on anyhow, so other than using strings I can't propagate context.

Reply:
This is now a public API, and returning String as Err is frowned upon (i think there's even a clippy lint for it)

> Currently aya does not depend on anyhow, so other than using strings I can't propagate context.

libraries should depend on thiserror, apps on anyhow. You can define a thiserror derived error, give it one variant for each string you're returning now, and type erase it when you return it. The result is the same as using a string (Error implements Display), but it's not a String, gives you a backtrace on nightly if you want one etc

Reply:
Done.